### DISCUSSION TITLE
Documentation: Unclear on how to set it up

### DISCUSSION THREAD

**Discussion by Rafee-M** [CogentRedTester/mpv-scripts#15] [Q&A] [answered]
Hi, I am a bit new to this. I moved all the scripts to `C:\Users\User1\AppData\Roaming\mpv\mpv\scripts` and I opened mpv. I can't seem to use all the features you've mentioned in the readme and I'd really appreciate it if you could tell me on how to set it up

![image](https://user-images.githubusercontent.com/69535896/151321352-cf141ab0-1436-4ced-ae01-beb49996918f.png)

**Comment by CogentRedTester** [comment]
All the scripts in this repository are separate with individual documentation at the top of each file.

**Comment by CogentRedTester** [comment, marked as answer]
Also, the base config folder mpv will use by default is `C:\Users\User1\AppData\Roaming\mpv\scripts`. If you've placed `mpv.exe` in that folder, then yes it could work, but using the `mpv` directoy next to the `exe` is considered very low priority and may be removed in the future. If the `exe` is in that folder you should either move it, or change the config directory to `C:\Users\User1\AppData\Roaming\mpv\portable_config\`.

**Comment by Rafee-M** [comment]
Thanks a lot for clearing that up! It works now.

**Comment by fuguesoft** [comment]
Debian 13.1 (trixie)
swayWM 1.21.0

Just installed the script and I can tell it's doing... something.
Whenever I try either google's API or Invidious I get a 403 Forbidden error.

```sh
[ffmpeg] https: HTTP error 403 Forbidden
Failed to open https://rr8---sn-bvvbaxivnuxq5uu-p5il.googlevideo.com/videoplayback?expire=1762183685&ei=pXUIacDGOOysy_sP35aduAU&ip=[IPADDRESS]&id=[REDACTED]&itag=251&source=youtube&requiressl=yes&xpc=[foo]&mm=31%2C29&mn=sn-bvvbaxivnuxq5uu-p5il%2Csn-q4flrnez&ms=au%2Crdu&mv=m&mvi=8&pl=42&rms=au%2Cau&initcwndbps=3280000&bui=[string]&spc=6b0G_FkOcR4ohYpXugmCPqyZfUN8t3QmllKSKAqBlLmut6m4fw&vprv=1&svpuc=1&mime=audio%2Fwebm&ns=RIpvr-rEesBE4lWoKJzshe8Q&rqh=1&gir=yes&clen=22106055&dur=1434.541&lmt=1761990700330557&mt=1762161689&fvip=5&keepalive=yes&lmw=1&fexp=51557447%2C51565115%2C51565681%2C51580970&c=TVHTML5&sefc=1&txp=4532534&n=YUMPvcLGb4McUPJ&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cns%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=[bar]&lsparams=cps%2Cmet%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=[baz].
EDL: Could not open source file 'https://rr8---sn-bvvbaxivnuxq5uu-p5il.googlevideo.com/videoplayback?expire=1762183685&ei=pXUIacDGOOysy_sP35aduAU&ip=[IPADDRESS]&id=[REDACTED]&itag=251&source=youtube&requiressl=yes&xpc=[foo]&mm=31%2C29&mn=sn-bvvbaxivnuxq5uu-p5il%2Csn-q4flrnez&ms=au%2Crdu&mv=m&mvi=8&pl=42&rms=au%2Cau&initcwndbps=3280000&bui=[string]&spc=6b0G_FkOcR4ohYpXugmCPqyZfUN8t3QmllKSKAqBlLmut6m4fw&vprv=1&svpuc=1&mime=audio%2Fwebm&ns=RIpvr-rEesBE4lWoKJzshe8Q&rqh=1&gir=yes&clen=22106055&dur=1434.541&lmt=1761990700330557&mt=1762161689&fvip=5&keepalive=yes&lmw=1&fexp=51557447%2C51565115%2C51565681%2C51580970&c=TVHTML5&sefc=1&txp=4532534&n=YUMPvcLGb4McUPJ&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cns%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=[bar]&lsparams=cps%2Cmet%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=[baz]'.
No video or audio streams selected.
Exiting... (Errors when loading file)
```

I'm running this from the command line and in both cases mpv refuses to open a normal youtube link.

I've set up my conf file in either of two ways but never both simultaneously:

### invidious

```conf
# The default config file for youtube-search
# Available at: https://github.com/CogentRedTester/mpv-scripts

# API key for the Google Youtube API
API_key=<my-api-key-without-quotes>

# number of search results to show in the list
num_results=40

# the url to send API calls to
API_path=https://invidious.f5.si/api/v1/
# use invidious API calls
invidious=yes

# attempt this API if the default fails
fallback_API_path=
fallback_invidious=yes

# the url to load videos from
frontend=https://invidious.f5.si/
```

### google

```conf
# The default config file for youtube-search
# Available at: https://github.com/CogentRedTester/mpv-scripts

# API key for the Google Youtube API
API_key=<my-api-key-without-quotes>

# number of search results to show in the list
num_results=40

# the url to send API calls to
API_path=https://www.googleapis.com/youtube/v3/

# use invidious API calls
invidious=no

# attempt this API if the default fails
fallback_API_path=https://googleapis.com/youtube/v3/
fallback_invidious=no

# the url to load videos from
frontend=https://www.youtube.com/
```

This is what my `$XDG_CONFIG_HOME/mpv/` looks like:

```sh
..
├── mpv.conf
├── script-modules
│   ├── scroll-list.lua
│   └── user-input-module.lua
├── script-opts
│   ├── mpv-clipper.conf
│   ├── SmartCopyPaste.conf
│   ├── streamsave.conf
│   ├── thumbfast.conf
│   └── youtube_search.conf
├── scripts
│   ├── mpv-clipper.lua
│   ├── SmartCopyPaste.lua
│   ├── streamsave.lua
│   ├── thumbfast.lua
│   ├── user-input.lua
│   └── youtube-search.lua
└── to_delete
    └── copy-permalink.lua

5 directories, 16 files
```

Before I added the API keys, I could at least get the search menu dialogue to
appear in mpv. Searching however, seemed to return no visible result. So I read
more of the config file parameters and attempted to configure it based on what I
could see.

After I did this, I notice that mpv could no longer play youtube links via: 
```sh
mpv <link>
```
It would always return 403 forbidden.

I have pinpointed the issue to be the `youtube_search.conf` script when
configured with the API settings as the lua script at least pretends to search
when I delete or otherwise reset the config to default.

What can I do?

Thanks for reading

E: Had to update yt-dlp. Disregard.
Very cool script! Thank you!

**Reply by fuguesoft** [reply]
Was able to get this script working with no API call
https://github.com/willswats/mpv-youtube-search
It just auto appends to or replaces the current playlist with the 50 or user defined number of videos it returns

E: Tried yours again after updating and changing the config to this

```conf
# The default config file for youtube-search
# Available at: https://github.com/CogentRedTester/mpv-scripts

# API key for the Google Youtube API
API_key=<my-api-key>

# number of search results to show in the list
num_results=40

# the url to send API calls to
API_path=https://inv.perditum.com/api/v1

# use invidious API calls
invidious=yes

# attempt this API if the default fails
fallback_API_path=
fallback_invidious=yes

# the url to load videos from
frontend=https://inv.perditum.com/
```
